barretenberg/cpp/src/barretenberg/crypto/merkle_tree/signal.hpp

```diff
@@ -47,17 +47,18 @@ class Signal {
      */
     void signal_level(uint32_t level = 0)
     {
-        signal_.store(level);
-        signal_.notify_all();
+        signal_->store(level);
+        signal_->notify_all();
     }
 
     void signal_decrement(uint32_t delta = 1)
     {
-        signal_.fetch_sub(delta);
-        signal_.notify_all();
+        signal_->fetch_sub(delta);
+        signal_->notify_all();
     }
 
   private:
-    std::atomic<uint32_t> signal_;
+    // apple clang has issues if this cannot be move-constructed, so we wrap in unique_ptr
```

Comment:
std::atomic seems a bit useless for storing in containers without a move constructor - wrap it